// keptn-provisioning/dist/api/schemes/resource-service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateResourceServiceScheme = void 0;
function generateResourceServiceScheme(project = '', stage = '', service = '', resourceURI = '') {
    return {
        baseURL: '/api/resource-service/v1',
        resources: {
            getListProjectResources: {
                url: `/project/${project}/stage/${stage}/service/${service}/resource`,
                method: 'get',
            },
            createServiceResources: {
                url: `/project/${project}/stage/${stage}/service/${service}/resource`,
                method: 'post',
            },
            updateServiceResources: {
                url: `/project/${project}/stage/${stage}/service/${service}/resource`,
                method: 'put',
            },
            deleteServiceResources: {
                url: `/project/${project}/stage/${stage}/service/${service}/resource/${encodeURIComponent(encodeURIComponent(resourceURI))}`,
                method: 'delete',
            },
        },
    };
}
exports.generateResourceServiceScheme = generateResourceServiceScheme;
